### ISSUE TITLE
Getting an "Invalid Date" during upload of 226 PDFs

### ISSUE THREAD

**Issue by brantje** [nextcloud/server#5714]
### Steps to reproduce
1. Use [this](https://thuisserver.ovh/s/URsqner3lvKPlOx) zip file, as this contains the pdf's i'm having trouble with.
2. Upload the files (or as folder)
3. After some while you will see "Invalid Date"
![qwdshov](https://user-images.githubusercontent.com/2758767/28168036-05519dfa-67de-11e7-926c-b28ab8611c90.png)

### Expected behaviour
No error message is given, nor a message on success

### Actual behaviour
Upload should succeed without any errors.

### Server configuration

**Operating system**: Ubuntu 16.04

**Web server:** Nginx

**Database:**: MariaDB

**PHP version:** 7.0

**Nextcloud version:** 12

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh 

**Where did you install Nextcloud from:** ZIP

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox

**Operating system:** Windows

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here

  | Doctrine\DBAL\Exception\UniqueConstraintViolationException: An
 exception occurred while executing 'INSERT INTO `oc_file_locks` 
(`key`,`lock`,`ttl`) SELECT ?,?,? FROM `oc_file_locks` WHERE `key` = ? 
HAVING COUNT(*) = 0' with params 
["files\/92b86e9233e0dc537b840e9cc8f3a875", 1, 1499928865, 
"files\/92b86e9233e0dc537b840e9cc8f3a875"]:

SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 
'files/92b86e9233e0dc537b840e9cc8f3a875' for key 'lock_key_index'
-- | --
Fatal | webdav | Sabre\DAV\Exception\BadRequest: HTTP/1.1 400 expected filesize 58769721 got 2195456 | 2017-07-13T15:11:50+0200
-- | -- | -- | --
Fatal | webdav | Sabre\DAV\Exception\BadRequest: HTTP/1.1 400 expected filesize 11382444 got 6742016 | 2017-07-13T15:06:00+0200
Fatal | webdav | OCA\DAV\Connector\Sabre\Exception\FileLocked: HTTP/1.1 423 "Documents/MS" is locked | 2017-07-13T15:05:47+0200
Fatal | webdav | Sabre\DAV\Exception\BadRequest: HTTP/1.1 400 expected filesize 41054610 got 40837120


```
</details>

**Comment by guiweber** [comment]
I'm seeing the same issue in NC 12.0.2 when uploading large files. The time left start by oscillating a lot between several minutes to hours, then at some point it just says "invalid date" and stays like this until the upload completes.

OS: Fedora 26
Server: Apache 2.4
PHP: 7.1.7
NC: 12.0.2
DB: MariaDB

**Comment by MorrisJobke** [comment]
Should be fixed in 12.0.4 and 13+: #6239
